src/librustdoc/clean/utils.rs

```diff
@@ -141,17 +141,15 @@ pub(super) fn external_path(
 }
 
 /// Remove the generic arguments from a path.
-crate fn strip_path_generics(path: Path) -> Path {
-    let segments = path
-        .segments
-        .iter()
-        .map(|s| PathSegment {
-            name: s.name,
+crate fn strip_path_generics(mut path: Path) -> Path {
+    for ps in path.segments.iter_mut() {
+        *ps = PathSegment {
+            name: ps.name,
```

Comment:
You can simplify this more by just writing to `ps.args` instead of creating a whole new struct.

Reply:
But i don't know what is inside that `ps.args`, how can i left it as it is?

Reply:
I don't understand your comment. I'm suggesting `ps.args = GenericArgs::AngleBracketed { args: vec![], bindings: vec![] }`, which doesn't require any knowledge about what's already there.

Reply:
I misread that as if suggestion was:
```rust
*ps = PathSegment {
    name: ps.name,
    args: ps.args,
}
```
Yes, this is simpler.